darli/modeling/integrators/fwd_euler.py

```diff
@@ -1,31 +1,20 @@
 """Module for forward euler integrator"""
-from .base import Integrator, StateSpaceBase, ArrayLike, cs
+from .base import Integrator, ModelBase, ArrayLike, cs
 
 
 class ForwardEuler(Integrator):
-    @staticmethod
+    def __init__(self, model: ModelBase):
+        super().__init__(model)
+
     def forward(
-        state_space: StateSpaceBase,
+        self,
+        derivative: ArrayLike,
```

Comment:
It is not an ArrayLike, rather Callable[[ArrayLike, ArrayLike, ArrayLike], ArrayLike]]
It is a function that maps (q, v, qfrc_u) -> (v, dv).

Once again, maybe it is enough to write Callable and in docstring describe what was really meant